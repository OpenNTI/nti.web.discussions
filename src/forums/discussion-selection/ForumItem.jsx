import React from 'react';
import PropTypes from 'prop-types';

import HighlightedContent from './HighlightedContent';


export default class ForumItem extends React.Component {
	static propTypes = {
		item: PropTypes.shape({
			title: PropTypes.string,
			get: PropTypes.func.isRequired
		}).isRequired,
		searchTerm: PropTypes.string,
		onClick: PropTypes.func,
	}

	onClick = () => this.props.onClick(this.props.item);

	render () {
		const { item, searchTerm } = this.props;

		return (
			<div className="discussion-selection-item" onClick={this.onClick}>
				<div className="content">
					<HighlightedContent content={item.get('displayTitle') || item.title} term={searchTerm} />
				</div>
				<div className="arrow-icon">
					<i className="icon-chevron-right" />
				</div>
			</div>
		);
	}
}
