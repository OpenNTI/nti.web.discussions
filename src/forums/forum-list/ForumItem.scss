@import '~@nti/style-common';

.forum-item-li {
	border-bottom: 1px dotted #d8d8d8;

	.forum-link {
		padding: .2rem 0 .2rem 0.9375rem;
		display: block;

		@include responsive-range(mobile) {
			padding: .625rem 0 .625rem 0.9375rem;
		}

		&.active {
			background: #3fb3f6;
			box-shadow: inset 3px 0 0 0 #107eed;
			.item-container {
				.item-main {
					.title {
						color: white;
					}

					.meta {
						color: white;
					}
				}

				.forum-item-edit {
					color: white;
				}
			}
		}
	}

	.item-container {
		display: flex;
		flex-direction: row;

		.item-main {
			flex-grow: 1;

			.title {
				display: block;
				font-size: rem-calc(14px);
				color: var(--primary-grey);
				font-family: "Open Sans Condensed";
				font-weight: 600;
				overflow: hidden;
				text-overflow: ellipsis;
				max-width: 200px;
				white-space: nowrap;

				@include responsive-range(mobile) {
					font-size: 1rem;
				}
			}

			.meta {
				font-size: .75rem;
				display: block;
				color: $tertiary-grey;
				font-family: 'Open Sans';
				font-weight: 600;

				@include responsive-range(mobile) {
					padding: .5rem 0;
				}
			}
		}

		.forum-item-reports {
			display: none;
			justify-content: center;
			align-items: center;
			padding: 0 rem-calc(5px);
			color: #ddd;
		}

		&:hover {
			.forum-item-reports {
				display: flex;
			}
		}
	}
}
