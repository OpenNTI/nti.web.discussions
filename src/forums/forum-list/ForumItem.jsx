import './ForumItem.scss';
import React from 'react';
import PropTypes from 'prop-types';
import {decorate} from '@nti/lib-commons';
import { LinkTo } from '@nti/web-routing';
import { HOC } from '@nti/web-commons';
import { scoped } from '@nti/lib-locale';
import { Viewer } from '@nti/web-reports';

const DEFAULT_TEXT = {
	count: {
		zero: 'No Discussions',
		one: '1 Discussion',
		other: '%(count)s Discussions'
	}
};

const t = scoped('forums.topic', DEFAULT_TEXT);

class ForumItem extends React.Component {
	static propTypes = {
		item: PropTypes.shape({
			displayTitle: PropTypes.string,
			title: PropTypes.string.isRequired,
			edit: PropTypes.func.isRequired,
			TopicCount: PropTypes.number.isRequired,
			hasLink: PropTypes.func.isRequired,
			Links: PropTypes.array.isRequired,
			Reports: PropTypes.array
		}).isRequired,
	}

	static contextTypes = {
		router: PropTypes.object
	}

	showReports = (e) => {
		e.preventDefault();

		const report = this.props.item.Reports[0];

		Viewer.show(report);
	}

	render () {
		const { item } = this.props;

		const canShowReports = item.Links.some(x => x.rel.startsWith('report-')) && item.Reports;

		return (
			<li className="forum-item-li">
				<LinkTo.Object object={item} activeClassName="active" className="forum-link">
					<div className="item-container">
						<div className="item-main">
							<span className="title">{item.displayTitle}</span>
							<div className="meta">
								<span className="see-all count">{t('count', { count: item.TopicCount })}</span>
							</div>
						</div>
						{canShowReports && (
							<div className="forum-item-reports" onClick={this.showReports}>
								<i className="icon-report" />
							</div>
						)}
					</div>
				</LinkTo.Object>
			</li>
		);
	}
}


export default decorate(ForumItem, [
	HOC.ItemChanges.compose
]);
