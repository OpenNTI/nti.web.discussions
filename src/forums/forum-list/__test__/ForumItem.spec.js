/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import ForumItem from '../ForumItem';

describe('Forum Item', () => {
	test('No Discussions - Snapshot', async () => {
		const item = {
			title: 'Forum',
			TopicCount: 0,
			hasLink: () => { return false; },
			edit: () => {},
			Links: []
		};
		const itemCmp = renderer.create(<ForumItem item={item} />);
		const tree = itemCmp.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
