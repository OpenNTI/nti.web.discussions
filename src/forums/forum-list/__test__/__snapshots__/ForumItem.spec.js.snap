// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forum Item No Discussions - Snapshot 1`] = `
<li
  className="forum-item-li"
>
  <div
    className="pathLink nti-link-to-path forum-link"
    to="#"
  >
    <div
      className="item-container"
    >
      <div
        className="item-main"
      >
        <span
          className="title"
        />
        <div
          className="meta"
        >
          <span
            className="see-all count"
          >
            No Discussions
          </span>
        </div>
      </div>
    </div>
  </div>
</li>
`;
